0.5.quality_control/0.jump_umap_analysis_with_cosmicqc.py

```diff
@@ -0,0 +1,698 @@
+            if color_dataframe is not None
+            else None
+        ),
+        cnorm="linear",
+        cmap=cmap,
+        bgcolor=bgcolor,
+        height=700,
+        width=800,
+        clabel=clabel,
+    )
+
+    # export the plot
+    hvplot.save(obj=plot, filename=filename, center=False)
+
+    return plot
+
+
+# +
+# form a path to the parquet file with single-cell profiles
+merged_single_cells = (
+    f"../0.download_data/data/plates/{example_plate}/{example_plate}.parquet"
+)
+
+# read only the metadata from parquet file
+parquet.ParquetFile(merged_single_cells).metadata
+# -
+
+# ## Process merged single-cell data using coSMicQC
+
+# + editable=true slideshow={"slide_type": ""}
+# show the first few columns for metadata column names
+schema_names = parquet.read_schema(merged_single_cells).names
+schema_names[:12]
+
+# +
+# set a list of metadata columns for use throughout
+metadata_cols = [
+    "Image_TableNumber",
+    "Metadata_ImageNumber",
+    "Image_Group_Number",
+    "Image_Group_Index",
+    "Image_Metadata_Col",
+    "Image_Metadata_Row",
+    "Image_Metadata_Site",
+    "Metadata_ObjectNumber",
+    "Metadata_Plate",
+    "Metadata_Well",
+    "Image_FileName_CellOutlines",
+    "Image_FileName_NucleiOutlines",
+    "Image_FileName_OrigAGP",
+    "Image_FileName_OrigDNA",
+    "Image_FileName_OrigRNA",
+    "Cytoplasm_AreaShape_BoundingBoxMaximum_X",
+    "Cytoplasm_AreaShape_BoundingBoxMaximum_Y",
+    "Cytoplasm_AreaShape_BoundingBoxMinimum_X",
+    "Cytoplasm_AreaShape_BoundingBoxMinimum_Y",
+]
+
+# read only metadata columns with feature columns used for outlier detection
+df_merged_single_cells = pd.read_parquet(
+    path=merged_single_cells,
+    columns=[
+        *metadata_cols,
+        "Nuclei_AreaShape_Area",
+        "Nuclei_AreaShape_FormFactor",
+        "Nuclei_AreaShape_Eccentricity",
+    ],
+)
+df_merged_single_cells.head()
+# -
+
+# show the original images and masks alignment with metadata
+df_merged_single_cells[
+    [
+        "Image_FileName_OrigAGP",
+        "Metadata_Well",
+        "Image_Metadata_Row",
+        "Image_Metadata_Col",
+        "Image_Metadata_Site",
+        "Image_FileName_CellOutlines",
+    ]
+]
+
+# +
+# create an outline and orig mapping dictionary to map original images to outlines
+# note: we turn off formatting here to avoid the key-value pairing definintion
+# from being reformatted by black, which is normally preferred.
+# fmt: off
+outline_to_orig_mapping = {
+    rf"{record['Metadata_Well']}_s{record['Image_Metadata_Site']}--cell_outlines\.png": 
+    rf"r{record['Image_Metadata_Row']:02d}c{record['Image_Metadata_Col']:02d}f{record['Image_Metadata_Site']:02d}p(\d{{2}})-.*\.tiff"  
+    for record in df_merged_single_cells[
+        [
+            "Metadata_Well",
+            "Image_Metadata_Row",
+            "Image_Metadata_Col",
+            "Image_Metadata_Site",
+        ]
+    ].to_dict(orient="records")
+}
```

Comment:
If we want to use nuclei outlines, would we just change that here? I recommend using nuclei outlines as the default as we tend to find outlier segmentation using nuclei mainly.

Reply:
Yes, I that's correct, I recall now we discussed moving over to nuclei over cell outlines. I'll make that change now.